tools/rostest/src/rostest/runner.py

```diff
@@ -121,7 +122,10 @@ def fn(self):
 
             #setup the test
             # - we pass in the output test_file name so we can scrape it
-            test_file = xmlResultsFile(test_pkg, test_name, False)
+            env = None
+            if results_base_dir:
+                env = {ROS_TEST_RESULTS_DIR: results_base_dir}
```

Comment:
Are there potentially other env values that this is going to override or suppress? It might be better to append to the env instead of overriding it.